app/templates/racer/index.html

```diff
@@ -76,10 +87,23 @@
   $('#season-slider').slider(isNoRaces ? 'disable' : 'enable');
   $('#fs-class-checkboxes').prop('disabled', isNoRaces);
 
-  var seasonRange = $('#season-slider').slider('getValue').join();
-  var classFilters = $('#class-checkboxes input')
+  var selectedClasses = $('#class-checkboxes input')
     .filter(function() { return this.checked; })
-    .map(function() { return this.value; }).get().join();
+    .map(function() { return this.value; }).get();
+
+  var allCkb = $('#class-checkbox-all')
+  if (selectedClasses.length != 0) {
+    allCkb.prop('checked', false);
+    allCkb.parent('label').removeClass('active');
+  } else {
+    allCkb.prop('checked', true);
+    allCkb.parent('label').addClass('active');
```

Comment:
@johnwheffner Is it worth marking the button as [disabled](http://getbootstrap.com/css/#buttons-disabled) when the state is also active? That would prevent a user from clicking the "All" button and expecting some type of result.

Thoughts?